src/lightning/fabric/strategies/fsdp.py

```diff
@@ -77,6 +77,12 @@ class FSDPStrategy(ParallelStrategy, _Sharded):
             checkpointing. This is typically your transformer block (including attention + feed-forward).
             Enabling this can free up a significant amount of memory at the cost of speed since activations in
             these layers need to be recomputed during backpropagation.
+        state_dict_type: The format in which the state of the model and optimizers gets saved into the checkpoint.
```

Comment:
can we think of more options than these two? Otherwise I'd make this a boolean flag of `sharded_state_dict` which needs less error handling (it's either true or false :D )

Reply:
I thought it's best to keep it aligned with the three types available in PyTorch FSDP: https://github.com/pytorch/pytorch/blob/c4752b1a9118b1ce18dfa10e255fe249507fd48f/torch/distributed/fsdp/api.py#L253-L255

There is a third one, the local state dict type. However, that one does not support retrieving the optimizer state at the moment, which is why we went for using the "sharded" type for saving and loading. I'm not sure what the future of local/sharded state dict type is in PyTorch. If we go with the current way here, we should be future-proof.